pkgs/by-name/ya/yarn-berry/fetcher/berry-4-offline.patch

```diff
@@ -0,0 +1,86 @@
+   async cloneFromRemote(locator: Locator, opts: FetchOptions) {
+     const repoUrlParts = gitUtils.splitRepoUrl(locator.reference);
+ 
+-    const cloneTarget = await gitUtils.clone(locator.reference, opts.project.configuration);
++    if (repoUrlParts.treeish.protocol !== "commit") {
++        throw new Error(`Missing source for git dependency ${locator.reference}`);
++    };
++
++    const cloneTarget = opts.cache.checkoutPath(repoUrlParts.treeish.request);
+     const projectPath = ppath.resolve(cloneTarget, repoUrlParts.extra.cwd ?? PortablePath.dot);
+ 
+     const packagePath = ppath.join(projectPath, `package.tgz`);
+diff --git a/packages/plugin-npm/sources/NpmSemverFetcher.ts b/packages/plugin-npm/sources/NpmSemverFetcher.ts
+index 7347859aa..ea5767f88 100644
+--- a/packages/plugin-npm/sources/NpmSemverFetcher.ts
++++ b/packages/plugin-npm/sources/NpmSemverFetcher.ts
+@@ -45,6 +45,7 @@ export class NpmSemverFetcher implements Fetcher {
+   }
+ 
+   private async fetchFromNetwork(locator: Locator, opts: FetchOptions) {
++    throw new Error(`Missing sources for ${structUtils.prettyLocator(opts.project.configuration, locator)}`);
+     let sourceBuffer;
+     try {
+       sourceBuffer = await npmHttpUtils.get(NpmSemverFetcher.getLocatorUrl(locator), {
+diff --git a/packages/yarnpkg-core/sources/Cache.ts b/packages/yarnpkg-core/sources/Cache.ts
+index b712ecf11..c7effbc61 100644
+--- a/packages/yarnpkg-core/sources/Cache.ts
++++ b/packages/yarnpkg-core/sources/Cache.ts
+@@ -225,6 +225,10 @@ export class Cache {
+     }
+   }
+ 
++  checkoutPath(commit: string): string {
++    return ppath.join(ppath.join(this.cwd, "../checkouts"), commit);
++  }
++
+   async fetchPackageFromCache(locator: Locator, expectedChecksum: string | null, {onHit, onMiss, loader, ...opts}: {onHit?: () => void, onMiss?: () => void, loader?: () => Promise<ZipFS>} & CacheOptions): Promise<[FakeFS<PortablePath>, () => void, string | null]> {
+     const mirrorPath = this.getLocatorMirrorPath(locator);
+ 
+diff --git a/packages/yarnpkg-core/sources/scriptUtils.ts b/packages/yarnpkg-core/sources/scriptUtils.ts
+index 2dcd7e59e..3079ad635 100644
+--- a/packages/yarnpkg-core/sources/scriptUtils.ts
++++ b/packages/yarnpkg-core/sources/scriptUtils.ts
+@@ -287,9 +287,9 @@ export async function prepareExternalProject(cwd: PortablePath, outputPath: Port
+             // Run an install; we can't avoid it unless we inspect the
+             // package.json, which I don't want to do to keep the codebase
+             // clean (even if it has a slight perf cost when cloning v1 repos)
+-            const install = await execUtils.pipevp(`yarn`, [`install`], {cwd, env, stdin, stdout, stderr, end: execUtils.EndStrategy.ErrorCode});
+-            if (install.code !== 0)
+-              return install.code;
++            //const install = await execUtils.pipevp(`yarn`, [`install`], {cwd, env, stdin, stdout, stderr, end: execUtils.EndStrategy.ErrorCode});
```

Comment:
It is probably easier to maintain for the future, if we delete the code instead of just commenting it.

Reply:
Same applies to `berry-3-offline.patch` .